src/vm/debughelp.cpp

```diff
@@ -476,7 +476,7 @@ int dumpStack(BYTE* topOfStack, unsigned len)
 
         if (isRetAddr((TADDR)*ptr, &whereCalled))
         {
-            if (_snwprintf_s(buffPtr, buff+NumItems(buff)-buffPtr-1, _TRUNCATE,  W("STK[%08X] = %08X "), (size_t)ptr, *ptr)  <0)
+            if (_snwprintf_s(buffPtr, buff + nLen - buffPtr - 1, _TRUNCATE,  W("STK[%08X] = %08X "), (size_t)ptr, *ptr)  <0)
```

Comment:
`- 1` should not be needed. The method takes the actual buffer size.

Also, could you please fix all places that take buffer size in this method to use the same expression while you are on it? It is a mix right now:

` wcsncpy_s(buffPtr, nLen - (buffPtr - buff)`
_snwprintf_s(buffPtr, buff+ nLen -buffPtr-1,`
`if( formatMethodTable(pMT, buffPtr, static_cast<DWORD>(buff+ nLen -buffPtr-1)) == NULL)`

Maybe add `buffEnd` variable and change all these to be just `buffEnd - buff`.